nixos/tests/pgadmin4.nix

```diff
@@ -5,62 +5,93 @@ import ./make-test-python.nix ({ pkgs, lib, ... }:
     pgadmin4Dir = "/var/lib/pgadmin";
     pgadmin4LogDir = "/var/log/pgadmin";
 
-    python-with-needed-packages = pkgs.python3.withPackages (ps: with ps; [
-      selenium
-      testtools
-      testscenarios
-      flask
-      flask-babelex
-      flask-babel
-      flask-gravatar
-      flask_login
-      flask_mail
-      flask_migrate
-      flask_sqlalchemy
-      flask_wtf
-      flask-compress
-      passlib
-      pytz
-      simplejson
-      six
-      sqlparse
-      wtforms
-      flask-paranoid
-      psutil
-      psycopg2
-      python-dateutil
-      sqlalchemy
-      itsdangerous
-      flask-security-too
-      bcrypt
-      cryptography
-      sshtunnel
-      ldap3
-      gssapi
-      flask-socketio
-      eventlet
-      httpagentparser
-      user-agents
-      wheel
-      authlib
-      qrcode
-      pillow
-      pyotp
-      boto3
-    ]);
   in
   {
     name = "pgadmin4";
     meta.maintainers = with lib.maintainers; [ gador ];
 
     nodes.machine = { pkgs, ... }: {
       imports = [ ./common/x11.nix ];
+      # needed because pgadmin 6.8 will fail, if those dependencies get updated
+      nixpkgs.overlays = [
```

Comment:
overriding is already done in the derivation, so why do it here as well

Reply:
See the comment below.

1) Pgadmin is not wrapped with test only python packages

2) Pgadmin (the app) is wrapped, but not the tests. To run the python tests, a python environment is needed.

Reply:
Please move this into the package possible behind a flag which builds it with test dependencies.

Reply:
This is already in the package in L74ff (`pgadmin4`). The override in the python package set need to be in the package, as well as here in the test. (Or do you know any way to re-use the `overrideScope` between derivations?)
 
Adding a test-flag to the pgadmin4 derivation is more work and more confusing than just adding the two test packages here in L71-L72 (`nixos/tests/pgadmin4`).

Reply:
If the overrideScope is required pgadmin should be not in pythonPackages and do them there. This will definitely bit rot over time and be forgotten when updating packages.

Reply:
What else would you suggest?
pgadmin will not start with the newer packaged versions and needs the older versions. Even if the overrides bit-rots and I forget about the overrides here: pgadmin is a leaflet packag and the tests will catch any build- and runtime errors due to the then-stale overrides.

Reply:
They need to be moved to ``pkgs/tools/admin/pgadmin/default.nix`` and only be in that file.

Reply:
So, if I understand you correctly, the `nixpkgs.overlays` must be removed here from the test and the `overrideScope` from the pgadmin.nix derivation must be imported here?

I really wrestled a long time with this (see https://github.com/NixOS/nixpkgs/pull/169370#discussion_r855898902) and this is the only way I got it to work. Any pointers of including the `overrideScope` from the pgadmin.nix into this test environment would be greatly appreciated.

Reply:
Ok, apparently I work better when its not late at night. I gave it a try to pass the python environment to the test from the derivation. Id appreciate a second (or third, forth...) look ;-)

Reply:
> So, if I understand you correctly, the `nixpkgs.overlays` must be removed here from the test and the `overrideScope` from the pgadmin.nix derivation must be imported here?

I took a second, deeper look and how it is right now not necessarily but if you would run the test standalone it wouldn't use the python env from pgadmin, so we still need to change it.

I think the easiest way would be to passthru the right things from pgadmin and use them in the test.

Reply:
Ok, I think I did that. 

The testcall to standalone is left as it is. The imported pgadmin package there is the one with the built-in python overrides.
The testcall to package is changed to pass the python environment. 

This way we only have to change it in one place.

Reply:
@SuperSandro2000
Do you have time for (yet) another look?